### PR TITLE
Add a new Api (AlertApi) and associated tests in the existing ApiService. 

### PR DESCRIPTION
**Pull request by mdivanova**
The new Api is not yet used: Added a comment in AlertApiService to say so.

### CODE REVIEW

File: backend/api-service/src/backend_firestore.go [hunk 1]
```diff
@@ -63,6 +63,10 @@ func (fb *FirestoreBackend) DeleteLocation(ctx context.Context, id string) error
 	return fb.deleteDoc(ctx, "locations", id)
 }
 
+func (fb *FirestoreBackend) DeleteAlert(ctx context.Context, id string) error {
+	return fb.deleteDoc(ctx, "alerts", id)
```
Comment (ssmall) [inline backend/api-service/src/backend_firestore.go:67]: Can this be pulled out to a `const` and reused?

File: backend/api-service/src/backend.go [hunk 2]
```diff
@@ -42,6 +42,7 @@ func (i *Inventory) applyTransaction(txn *InventoryTransaction) error {
 type DatabaseBackend interface {
 	DeleteItem(ctx context.Context, id string) error
 	DeleteLocation(ctx context.Context, id string) error
+	DeleteAlert(ctx context.Context, id string) error
```
Comment (josueetcom) [inline backend/api-service/src/backend.go:45]: Nit: could these be kept in alphabetical order?

File: backend/api-service/src/backend_memory_test.go [hunk 3]
```diff
@@ -64,6 +67,14 @@ func TestIMBDeleteLocationNotFound(t *testing.T) {
 	inMemoryBackendTester.testDeleteLocationNotFound(t)
 }
 
+func TestIMBDeleteAlert(t *testing.T) {
+	inMemoryBackendTester.testDeleteAlert(t)
+}
+
+func TestIMBDeleteAmertNotFound(t *testing.T) {
```
Comment (josueetcom) [inline backend/api-service/src/backend_memory_test.go:74]: *Alert

File: backend/api-service/src/backend_memory_test.go [hunk 4]
```diff
@@ -116,8 +127,8 @@ func TestIMBListLocationInventoryTransactions(t *testing.T) {
 	inMemoryBackendTester.testListLocationInventoryTransactions(t)
 }
 
-func TestIMBNewItem(t *testing.T) {
-	inMemoryBackendTester.testNewItem(t)
```
Comment (josueetcom) [inline backend/api-service/src/backend_memory_test.go:120]: Why was this deleted?

File: backend/api-service/src/backend_tester.go [hunk 5]
```diff
@@ -124,6 +127,48 @@ func (bt *backendTester) testDeleteLocationNotFound(t *testing.T) {
 	}
 }
 
+func (bt *backendTester) testDeleteAlert(t *testing.T) {
+	ctx := context.Background()
+	id := "alert-id"
+	alert := Alert{
+		Id:            "id-to-be-replaced-by-uuid",
```
Comment (josueetcom) [inline backend/api-service/src/backend_tester.go:134]: nit: technically this won't be replaced by a uuid as it's part of the initial backend state

File: backend/api-service/src/backend_firestore.go (hunk 1 above)
Comment (mdivanova) [reply backend/api-service/src/backend_firestore.go]: Yep! I pulled out the strings for the other objects too to be consistent.

File: backend/api-service/src/backend.go (hunk 2 above)
Comment (mdivanova) [reply backend/api-service/src/backend.go]: Yep - done.

File: backend/api-service/src/backend_memory_test.go (hunk 3 above)
Comment (mdivanova) [reply backend/api-service/src/backend_memory_test.go]: thanks!

File: backend/api-service/src/backend_memory_test.go (hunk 4 above)
Comment (mdivanova) [reply backend/api-service/src/backend_memory_test.go]: good catch - thank you! I don't know how that happened.

File: backend/api-service/src/backend_tester.go (hunk 5 above)
Comment (mdivanova) [reply backend/api-service/src/backend_tester.go]: Changed to just "id".

File: backend/api-service/src/backend_firestore.go [hunk 6]
```diff
@@ -24,6 +24,12 @@ import (
 	"google.golang.org/grpc/status"
 )
 
+const alertsStr = "alerts"
```
Comment (ssmall) [inline backend/api-service/src/backend_firestore.go:27]: nit:  it would be useful to name these after what they represent (Firestore collections), so perhaps `alertsCollection` would be a more appropriate name.

File: backend/api-service/src/backend_firestore.go [hunk 7]
```diff
@@ -24,6 +24,12 @@ import (
...
+const inventoriesStr = "inventories"
+const inventoryTransactionsStr = "inventoryTransactions"
+const itemsStr = "items"
+const locationsStr = "locations"
```
Comment (ssmall) [inline backend/api-service/src/backend_firestore.go:31]: nit: could be more succinctly written as 
```go
const (
  alertsStr = "alerts"
  inventoriesStr = "inventories"
  // etc
)
```

File: backend/api-service/src/backend_firestore.go (hunk 6 above)
Comment (mdivanova) [reply backend/api-service/src/backend_firestore.go]: thanks - renamed as per the suggestion.

File: backend/api-service/src/backend_firestore.go (hunk 7 above)
Comment (mdivanova) [reply backend/api-service/src/backend_firestore.go]: Done.

Comment (zsxking) [reply backend/api-service/src/backend_firestore.go]: As go readability recommend, it's better to avoid repeating the name of the type of a variable in its name. So it's better to omit the `Str` suffix in these names.

Comment (mdivanova) [reply backend/api-service/src/backend_firestore.go]: I renamed to "Collection" from another comment from Spencer. There are already other variables with the name without any suffix so I can't just remove it, e.g. (locations).
